### PR TITLE
DeMarcus Kirby

### PR DESCRIPTION
**Pull request by KirbyDD**
Last test not passing

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @revenue = 0
+  end
+
+  def add_exhibit exhibit
+    @exhibits << exhibit
+  end
+
+  def recommend_exhibits patron
+    result = []
+    @exhibits.each do |exhibit|
```
Comment (aziobrow) [inline lib/museum.rb]: if you're using `each` to shovel, a `map` is probably a good tool to consider 😉

Comment (aziobrow) [reply lib/museum.rb]: Niiice work figuring out a nice solution to a difficult problem.

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def patrons_by_exhibit_interest
+    hash = @exhibits.reduce({}) do |acc,exhibit|
+      acc[exhibit] = []
+      acc
+    end
+
+    @exhibits.each do |exhibit|
+      @patrons.each do |patron|
```
Comment (aziobrow) [inline lib/museum.rb]: You can do all of this inside of the reduce, but again, nice job getting this to work. `reduce` is already an enumerable, so there's no reason to reduce over exhibits and then re-enumerate with `each.`

File: lib/museum.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    hash
+  end
+
+  def attend_exhibits patron
+    @exhibits.each do |exhibit|
+      if patron.interests.include? exhibit.name && patron.spending_money >= exhibit.cost
+        patron.spending_money -= exhibit.cost
+      end
+    end
+  end
```
Comment (aziobrow) [inline lib/museum.rb]: Whoa. This took me a while to debug...I saw it didn't pass the test, but it looked correct in terms of logic, so I dug in... 

You ran into a very tricky combination of precedence (remember Ruby order of operations) and bad luck in choosing not to use parentheses... 

`patron.interests.include? exhibit.name && patron.spending_money >= exhibit.cost`
So >= is evaluated first in precedence, so it's really splitting it up into:
`patron.interests.include? exhibit.name && patron.spending_money` >=`exhibit.cost`

So then it's evaluating the left hand side, and since you didn't use parentheses around `exhibit.name`, it thinks `exhibit.name && patron.spending_money` is ALL the argument for `include?`, so it's checking `patron.interests` for a combination of the exhibit name and the amount of spending money, which it's obviously never going to find. So it's always going to return false.   

So ultimately, the left hand side of the equation (to the left of >=) is always going to return false, and even though the right hand side is returning truthy (since `patron.spending_money` isn't nil or false), false && true always returns false. So the -= line is never executing. 

tl;dr--- Be really careful when you use multiple boolean operators in the same line, and probably just generally use parentheses with methods that require arguments, unless that method is the only thing on the line. (and maybe just use them anyway, since they make code a lot more readable)

I know this is a really suble, tricky one, so let me know if it doesn't make sense.
